packages/styled/hooks/useStandardProps.js

```diff
@@ -0,0 +1,90 @@
+import { useContext } from 'react';
+import classNames from 'classnames';
+import get from 'lodash/get';
+import memoize from 'lodash/memoize';
+import { ThemeContext } from 'styled-components';
+
+/**
+ * Replace relevant style attribute values with values from site theme.
+ *
+ * @param {object} styleObject Object corresponding to style attribute.
+ * @param {object} siteTheme Theme object passed in via React Context through Styled Components
+ *   <ThemeProvider /> (https://styled-components.com/docs/advanced#theming).
+ */
+export const replaceWithSiteTheme = (styleObject, siteTheme) => (
+  Object.keys(styleObject)
+    .reduce((acc, property) => {
+      /**
+       * For every value passed in the style prop, check if the value is
+       * actually a path to our ThemeContext.
+       *
+       * Example,
+       *   Match:
+       *    {color: theme.colors.black} => {color: #000;}
+       *
+       *   No Match:
+       *     {color: #000} => {color: #000;}
+       */
+      const themeStyle = get(
+        siteTheme,
+        styleObject[property],
+        styleObject[property]
+      );
+
+      return {
+        ...acc,
+        [property]: themeStyle,
+      };
+    }, styleObject)
+);
+
+export const createComponentNameClass = memoize((componentName) => (
+  componentName
+    .replace(/(\/|\\)/, '__') // replace any back- or forward-slashes with double underscores.
+    .replace(/[^_a-zA-Z0-9-]/g, '-') // replace any invalid character with a hyphen.
+));
+
+/**
+ * Provide a component with a standardized set of props for that component to use.
+ * @param {object} props    React comonent props.
+ * @param {object} componentDefaults Overrides for default prop values.
+ */
+const useStandardProps = (props, componentDefaults = {}) => {
+  const {
+    className = '',
+    id = '',
+    style = {},
+    tag = '',
```

Comment:
I'm unsure if `tag` really belongs here. I would assume that `standardProps` should be able to be applied to most, if not all components. Can you pass a tag to most of our core components in order to transform the generated markup?

Reply:
I'm a bit on the fence here, but I lean toward leaving it in because this is in the `styled` package, which assumes you're planning to primarily use styled components for your styling.